src/OrleansCodeGenerator/OrleansCodeGenerator.csproj

```diff
@@ -1,82 +1,30 @@
-    <DefineConstants>DEBUG;TRACE</DefineConstants>
-    <ErrorReport>prompt</ErrorReport>
-    <WarningLevel>4</WarningLevel>
-    <DocumentationFile>bin\Debug\OrleansCodeGenerator.XML</DocumentationFile>
+    <PackageId>Microsoft.Orleans.OrleansCodeGenerator</PackageId>
+    <Title>Microsoft Orleans Code Generation</Title>
+    <Description>Runtime and compile-time code generation support for Microsoft Orleans.</Description>
   </PropertyGroup>
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
-    <DebugType>pdbonly</DebugType>
-    <Optimize>true</Optimize>
-    <OutputPath>bin\Release\</OutputPath>
-    <DefineConstants>TRACE</DefineConstants>
-    <ErrorReport>prompt</ErrorReport>
-    <WarningLevel>4</WarningLevel>
-    <DocumentationFile>bin\Release\OrleansCodeGenerator.XML</DocumentationFile>
-  </PropertyGroup>
-  <PropertyGroup Condition="'$(Bootstrap)' == 'true'">
-    <OutputPath>$(BootstrapOutputPath)</OutputPath>
-    <OutDir>$(BootstrapOutputPath)</OutDir>
-    <IntermediateOutputPath>$(BootstrapOutputPath)Intermediate\</IntermediateOutputPath>
-    <DefineConstants>$(DefineConstants);EXCLUDE_CODEGEN</DefineConstants>
+
+  <PropertyGroup>
+    <AssemblyName>OrleansCodeGenerator</AssemblyName>
+    <RootNamespace>OrleansCodeGenerator</RootNamespace>
   </PropertyGroup>
+
   <ItemGroup>
-    <Reference Include="System" />
-    <Reference Include="System.Core" />
-    <Reference Include="Microsoft.CSharp" />
+    <ProjectReference Include="..\Orleans\Orleans.csproj" />
   </ItemGroup>
+
   <ItemGroup>
-    <Compile Include="..\Build\GlobalAssemblyInfo.cs">
-      <Link>Properties\GlobalAssemblyInfo.cs</Link>
-    </Compile>
-    <Compile Include="GeneratedSyntax.cs" />
-    <Compile Include="SerializerGenerationManager.cs" />
-    <Compile Include="SerializerGenerator.cs" />
-    <Compile Include="RoslynCodeGenerator.cs" />
-    <Compile Include="CodeGenerationException.cs" />
-    <Compile Include="CodeGeneratorCommon.cs" />
-    <Compile Include="GrainMethodInvokerGenerator.cs" />
-    <Compile Include="GrainReferenceGenerator.cs" />
-    <Compile Include="Properties\AssemblyInfo.cs" />
-    <Compile Include="Utilities\StringExtensions.cs" />
-    <Compile Include="Utilities\SyntaxFactoryExtensions.cs" />
+    <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="$(MicrosoftCodeAnalysisCSharpVersion)" />
   </ItemGroup>
-  <ItemGroup>
-    <ProjectReference Include="..\Orleans\Orleans.csproj">
-      <Project>{bc1bd60c-e7d8-4452-a21c-290aec8e2e74}</Project>
-      <Name>Orleans</Name>
-    </ProjectReference>
+
+  <ItemGroup Condition=" '$(BuildFlavor)' == 'Current' ">
+    <ProjectReference Include="..\Orleans.PlatformServices\Orleans.PlatformServices.csproj" />
   </ItemGroup>
-  <ItemGroup>
-    <None Include="project.json" />
+
+  <ItemGroup Condition=" '$(BuildFlavor)' == 'Current' ">
+    <PackageReference Include="System.Reflection" Version="$(SystemRuntimeVersion)" />
```

Comment:
wouldn't it be better to have a SystemReflectionVersion variable defined in the props file too? That one can also point (for now) to SystemRuntimeVersion if they do align, but I think it's better than then trying to find all places where you used SystemRuntimeVersion but you meant a different package